--- libsyncml-0.5.4/libsyncml/parser/sml_wbxml_internals.h~	2009-06-04 18:16:56.000000000 +0900
+++ libsyncml-0.5.4/libsyncml/parser/sml_wbxml_internals.h	2015-08-26 23:01:34.743734099 +0900
@@ -29,7 +29,7 @@
 #ifndef _SML_WBXML_INTERNALS_H_
 #define _SML_WBXML_INTERNALS_H_
 
-#include <wbxml.h>
+#include <wbxml/wbxml.h>
 
 SmlBool smlWbxmlConvertTo(WBXMLConvXML2WBXMLParams *params, const char *input, char **output, size_t *outputLen, SmlError** error);
 SmlBool smlWbxmlConvertFrom(WBXMLConvWBXML2XMLParams *params, const char *input, size_t inputLen, char **output, size_t *outputLen, SmlError** error);
--- libsyncml-0.5.4/tests/mobiles/obex_mobile_ds_client.c~	2009-06-04 18:16:56.000000000 +0900
+++ libsyncml-0.5.4/tests/mobiles/obex_mobile_ds_client.c	2015-08-26 23:07:31.475606868 +0900
@@ -28,7 +28,7 @@
  */
 
 #include <libsyncml/sml_transport_internals.h>
-#include <wbxml.h>
+#include <wbxml/wbxml.h>
 
 /* external configuration */
 
